components/brave_rewards/browser/rewards_service_impl.h

```diff
@@ -720,6 +722,10 @@ class RewardsServiceImpl : public RewardsService,
 
   GetTestResponseCallback test_response_callback_;
 
+  // At the moment we keep it only for the purpose of sending P3A stats.
+  // Used only on UI thread.
+  bool auto_contributions_enabled_ = false;
```

Comment:
could we just use `GetAutoContribute` function so that we don't need to introduce this state?

Reply:
that would make this more difficult https://github.com/brave/brave-core/pull/3242/files#diff-a5d16be9c9b52a6df7b866b1d75d3f5fR3812 - but I'm not sure it makes sense to send that value to the db method. Why call `RecordAutoContributionsState` inside a db metrics call instead of just calling it on init/when the value changes?

Reply:
the metric is somewhat complex as per spec - to record the value we need both this flag and number of contributions. The number of contributions is available only on DB thread so I have to pass the flag there.